src/cautiousrobot/utils.py

```diff
@@ -18,3 +21,27 @@ def update_log(log, index, filepath):
     with open(filepath, "a") as log_file:
         json.dump(log[index], log_file, indent = 4)
         log_file.write("\n")
+
+
+def process_csv(csv_path, expected_cols):
+    '''
+    Reads a CSV, sets all columns to lowercase (for case-insensitivity, and checks for expected columns.
+    
+    Parameters:
+    csv_path - String. Path to the CSV.
+    expected_cols - Dictionary of column headers expected to be present in the CSV.
+
+    Returns:
+    df - DataFrame processed from the CSV.
+    '''
+    df = pd.read_csv(csv_path, low_memory = False)
+    df.columns = df.columns.str.lower()
+
+    missing_cols = []
+    for col in list(expected_cols.keys()):
+        if expected_cols[col] not in list(df.columns):
+            missing_cols.append(col)
+    if len(missing_cols) > 0:
+        sys.exit(f"The CSV at {csv_path} is missing column(s): {missing_cols}, defined as {[expected_cols[col] for col in missing_cols]}")
```

Comment:
Consider throwing an exception instead. That would make this function more flexible since sys.exit will terminate the program.

Reply:
You mean throw an exception and give an option for amended input? Or just have this function throw an exception and the main program can exit or ask for revised input as a result?

Reply:
I would have the function throw an exception and catch it higher up. Having a function named `process_csv` terminate the entire python process if something is wrong is rather unexpected.